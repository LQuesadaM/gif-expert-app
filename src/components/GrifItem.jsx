import PropTypes from 'prop-types'

export const GrifItem = ({ title, url, id }) => {
	return (
		<div className="card">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	)
}

GrifItem.propType = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
}
