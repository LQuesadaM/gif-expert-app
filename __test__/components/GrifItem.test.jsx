import { render, screen } from '@testing-library/react'
import { GrifItem } from '../../src/components/GrifItem.jsx'

describe('Prueba en <GriftItem/>', () => {
	const title = 'Saitama'
	const url = 'https://one-puch-man.com/saitama.jpg'
	test('Debe de hacer match con el snapshop', () => {
		const { container } = render(<GrifItem title={title} url={url} />)
		expect(container).toMatchSnapshot()
	})

	test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
		render(<GrifItem title={title} url={url} />)
		//expect(screen.getByRole('img').src).toBe(url)
		//expect(screen.getByRole('img').alt).toBe(title)
		const { src, alt } = screen.getByRole('img')
		expect(src).toBe(url)
		expect(alt).toBe(title)
	})

	test('Debe de mostrar el titulo en el componente', () => {
		render(<GrifItem title={title} url={url} />)
		expect(screen.getByText(title)).toBeTruthy()
	})
})
