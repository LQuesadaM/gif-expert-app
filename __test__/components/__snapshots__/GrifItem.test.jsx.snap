// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GriftItem/> Debe de hacer match con el snapshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-puch-man.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
